### ISSUE TITLE
*gettext macros fail with string sigils

### ISSUE THREAD

**Issue by whatyouhide** [elixir-lang/gettext#57]
```elixir
import MyApp.Gettext
gettext ~s(foo)
```
 
fails with `** (ArgumentError) msgid must be a string literal`.

**Comment by josevalim** [comment]
Weird, I thought we macro expanded them?

**Comment by whatyouhide** [comment]
We do, I'm not sure why this happens. I'll investigate!

**Comment by whatyouhide** [comment]
@josevalim ok it's because

```elixir
Macro.expand (quote do: ~s(foo)), __ENV__
#=> {:<<>>, [], ["foo"]}
```

I can manually check for the `{:<<>>, _, [binary]}` pattern.

**Comment by josevalim** [comment]
Yes, we need to check for `:<<>>` as long as all entries are binaries. I will update sigil_s to return a binary if such is guaranteed.

**Comment by whatyouhide** [comment]
Closed by 2f183e33ce8ac7ff317b255eb27c809526eb9535! /cc @josevalim
